pkg/healthcheck/healthcheck_test.go

```diff
@@ -0,0 +1,355 @@
+package healthcheck
+
+import (
+	"encoding/json"
+	"errors"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/vishvananda/netlink"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	fakecorev1 "k8s.io/client-go/kubernetes/typed/core/v1/fake"
+	clientgotesting "k8s.io/client-go/testing"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+)
+
+const TESTHOSTNAME = "worker"
+
+var (
+	fakeNodesJson = `{"items":[{"metadata":{"name":"` + TESTHOSTNAME + `"},"spec":{"taints":[{"effect":"NoSchedule",
+					"key":"` + TAINT_KEY + `"}]}}]}`
+	fakeConf        = "interfaces: ['ifA', 'ifB']\nreachability:\n- host: someAddress\n  port: 42\ntimeout: 5s"
+	fakeConfInvalid = "-invalid"
+	fakeNodes       *v1.NodeList
+	tmpPath         string
+)
+
+var _ = BeforeSuite(func() {
+	fakeNodes = &v1.NodeList{}
+	err := json.Unmarshal([]byte(fakeNodesJson), fakeNodes)
+	Expect(err).ShouldNot(HaveOccurred())
+})
+
+func TestHealthCheck(t *testing.T) {
+	RegisterFailHandler(Fail)
+	t.Setenv(NODE_NAME_ENV, TESTHOSTNAME)
+	tmpPath = t.TempDir()
+	RunSpecsWithDefaultAndCustomReporters(t,
+		"HealthCheck Suite",
+		[]Reporter{printer.NewlineReporter{}})
+}
+
+var _ = Describe("HealthCheck", func() {
+	var _ = Context("LoadConfig() should", func() {
+		var _ = It("return error if config is mandatory but does not exist", func() {
+			t := testing.T{}
+			t.Setenv(CONFIG_ENV, "/some/invalid/path")
+			_, err := LoadConfig("")
+			Expect(err).To(HaveOccurred())
+			t.Setenv(CONFIG_ENV, "")
+		})
+		var _ = It("return error if config is invalid", func() {
+			filePath, err := os.CreateTemp(tmpPath, "healtchecktest")
+			Expect(err).ToNot(HaveOccurred())
+			_, err = filePath.WriteString(fakeConfInvalid)
+			Expect(err).ToNot(HaveOccurred())
+			_, err = LoadConfig(filePath.Name())
+			Expect(err).To(HaveOccurred())
+		})
+		var _ = It("return no error if valid config exists", func() {
+			filePath, err := os.CreateTemp(tmpPath, "healtchecktest")
+			Expect(err).ToNot(HaveOccurred())
+			_, err = filePath.WriteString(fakeConf)
```

Comment:
for the fakeconf you could also just create a `testdata` directory in this package, put it in there and commit to git, and then load using a relative path

Reply:
Done

Did not want to add irrelevant files to the repository, but if you say it's OK, then it's simpler approach, which I like.